### PR TITLE
[W5.7][T12-3]Ang Chen Shin

### PR DESCRIPTION
**Pull request by CrimsonAng**
Added Count command

### CODE REVIEW

**Comment by CrimsonAng** [comment]
Self-review. Find out why Travis CI build failed even though code passed the tests inside IntelliJ IDEA

**Comment by hotarumin** [comment]
why did you remove unnecessary imports? and how did you know which ones were unnecessary?

**Comment by thaddeuslzy** [comment]
Good to see that your code now passes the Travis CI build test. But i'm not sure how the changes caused it to work!

File: src/seedu/addressbook/commands/CountCommand.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+    @Override
+    public CommandResult execute() {
+        List<ReadOnlyPerson> allPersons = addressBook.getAllPersons().immutableListView();
+        return new CommandResult(getMessageForPersonListShownSummary(allPersons)/*, allPersons*/);
+    }
+}
+
+
+
```
Comment (parvathy-sudhir-pillai) [inline src/seedu/addressbook/commands/CountCommand.java:27]: Stray line breaks? Check diff before submitting PR

File: test/expected.txt [hunk 2]
```diff
@@ -290,6 +292,9 @@
 || 
 || 2 persons listed!
 || ===================================================
+|| Enter command: || [Command entered:  count]
+|| 2 persons listed!
+|| ===================================================
```
Comment (parvathy-sudhir-pillai) [inline test/expected.txt:297]: Good job updating the test i/o.

File: test/java/seedu/addressbook/commands/CountCommandTest.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+import org.junit.Test;
+
+import seedu.addressbook.data.AddressBook;
+import seedu.addressbook.data.person.ReadOnlyPerson;
+import seedu.addressbook.util.TestUtil;
+
+import java.util.Collections;
+import java.util.List;
+
+public class CountCommandTest {
```
Comment (parvathy-sudhir-pillai) [inline test/java/seedu/addressbook/commands/CountCommandTest.java:14]: Public classes should have descriptions; helps generate java docs

File: test/java/seedu/addressbook/commands/CountCommandTest.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    @Test
+    public void countCheckNonZero() throws Exception {
+
+        AddressBook addressBook = new AddressBook();
+        addressBook.addPerson(TestUtil.generateTestPerson());
+
+        CountCommand count = new CountCommand();
+        count.setData(addressBook, addressBook.getAllPersons().immutableListView());
+
+        assertEquals("1 persons listed!", count.execute().feedbackToUser);
```
Comment (parvathy-sudhir-pillai) [inline test/java/seedu/addressbook/commands/CountCommandTest.java:26]: Could also include jUnit tests to check a few more cases esp; when it is issued for an empty AddressBook.

File: src/seedu/addressbook/commands/CountCommand.java [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Count all persons in the address book and display the result.\n"
+            + "Example: " + COMMAND_WORD;
+
+
+    @Override
+    public CommandResult execute() {
+        List<ReadOnlyPerson> allPersons = addressBook.getAllPersons().immutableListView();
+        return new CommandResult(getMessageForPersonListShownSummary(allPersons)/*, allPersons*/);
```
Comment (parvathy-sudhir-pillai) [inline src/seedu/addressbook/commands/CountCommand.java:22]: Don't comment out code. We use version control precisely to go back to a previous version. Just go ahead and delete the commented out code.

**Review by parvathy-sudhir-pillai — APPROVED** [review]
Good effort on this PR @CrimsonAng. In particular, good job on,
* updating docs
* updating I/O tests
* keeping the PR relatively clean

Suggestions to improve further:
* update developer docs
* more jUnit tests
* Pay closer attention to the coding standard

Please close the PR after viewing comments.
